src/monster.cpp

```diff
@@ -1648,9 +1648,12 @@ void monster::drop_items_on_death()
     if( !type->in_species( ZOMBIE ) ) {
         return;
     }
-    for( const auto &it : dropped ) {
-        if( it->is_armor() ) {
-            it->item_tags.insert( "FILTHY" );
+
+    if( !ACTIVE_WORLD_OPTIONS[ "NO_FILTHY" ] ) {
```

Comment:
The mod should either blacklist the squeamish trait or leave filthiness on clothing and just zero the morale.
At the moment it disables filthiness but doesn't touch squeamish, turning it into a free point.